app/controllers/sessions_controller.rb

```diff
@@ -1,6 +1,11 @@
 class SessionsController < ApplicationController
   def new
-
+    if current_user != nil
+      flash[:message] = "You are already logged in."
+      redirect_to profile_path if current_user.role == "user"
```

Comment:
Chris, I don't know if you are aware but we do have helper methods `current_merchant?` and `current_admin?`. I think these can replace `current_user.role == "merchant"` and so so.

Reply:
Agreed, changing on branch.